### ISSUE TITLE
Zabbix agent not exposing its port 10050

### ISSUE THREAD

**Issue by aqiank** [zabbix/zabbix-docker#813]
Hi, I'm using the `docker-compose_v3_alpine_pgsql_latest.yaml` file from tag `5.2.5` and it seems like everything runs except that Zabbix Server says that it cannot reach the Zabbix Agent. When I see `docker ps`, I noticed that its port 10050 is not exposed which I guess is why it doesn't work. This is the partial output of `docker ps` containing the Zabbix Agent:
```
722372e65167   zabbix/zabbix-agent:alpine-5.2-latest              "/sbin/tini -- /usr/…"   27 hours ago   Up 26 hours                                                              zabbix-docker_zabbix-agent_1
```
If that port not being exposed is the cause, then I'm not sure what I should do to fix it. The online thing that I have changed is the ZBX_SERVER_NAME in `.env_web` file. Do you guys have any idea what is going on? Thanks!

**Comment by rockaut** [comment]
Have you checked this #809 ?

**Comment by aqiank** [comment]
Hi @rockaut, Yeah I have checked that issue. I wasn't able to get information that could help my case from it. It seems like he was able to get port 10050 exposed but I'm not able to.

**Comment by rockaut** [comment]
Ok - more infos needed :D

Are the containers running on the same host? If so you don't even need to expose the agent ports. Container to container in the same container-network works without exposing.

Can you post your docker-compose file please? And I suppose you're using Docker? On Windows?

**Comment by aqiank** [comment]
The docker-compose file is the same as the one in Github (I have put it below, just in case). I created a symlink `docker-compose.yaml` that points to `docker-compose_v3_alpine_pgsql_latest.yaml` and use `docker-compose up -d` to start everything up. Here is the docker-compose file:
```
version: '3.5'
services:
 zabbix-server:
  image: zabbix/zabbix-server-pgsql:alpine-5.2-latest
  ports:
   - "10051:10051"
  volumes:
   - /etc/localtime:/etc/localtime:ro
   - /etc/timezone:/etc/timezone:ro 
   - ./zbx_env/usr/lib/zabbix/alertscripts:/usr/lib/zabbix/alertscripts:ro
   - ./zbx_env/usr/lib/zabbix/externalscripts:/usr/lib/zabbix/externalscripts:ro
   - ./zbx_env/var/lib/zabbix/export:/var/lib/zabbix/export:rw
   - ./zbx_env/var/lib/zabbix/modules:/var/lib/zabbix/modules:ro
   - ./zbx_env/var/lib/zabbix/enc:/var/lib/zabbix/enc:ro
   - ./zbx_env/var/lib/zabbix/ssh_keys:/var/lib/zabbix/ssh_keys:ro
   - ./zbx_env/var/lib/zabbix/mibs:/var/lib/zabbix/mibs:ro
   - ./zbx_env/var/lib/zabbix/snmptraps:/var/lib/zabbix/snmptraps:ro
#   - ./.ZBX_DB_CA_FILE:/run/secrets/root-ca.pem:ro
#   - ./.ZBX_DB_CERT_FILE:/run/secrets/client-cert.pem:ro
#   - ./.ZBX_DB_KEY_FILE:/run/secrets/client-key.pem:ro
  links:
   - postgres-server:postgres-server
   - zabbix-java-gateway:zabbix-java-gateway
  ulimits:
   nproc: 65535
   nofile:
    soft: 20000
    hard: 40000
  deploy:
   resources:
    limits:
      cpus: '0.70'
      memory: 1G
    reservations:
      cpus: '0.5'
      memory: 512M
  env_file:
   - .env_db_pgsql
   - .env_srv
  secrets:
   - POSTGRES_USER
   - POSTGRES_PASSWORD
  depends_on:
   - postgres-server
   - zabbix-java-gateway
   - zabbix-snmptraps
  networks:
   zbx_net_backend:
     aliases:
      - zabbix-server
      - zabbix-server-pgsql
      - zabbix-server-alpine-pgsql
      - zabbix-server-pgsql-alpine
   zbx_net_frontend:
#  devices:
#   - "/dev/ttyUSB0:/dev/ttyUSB0"
  stop_grace_period: 30s
  sysctls:
   - net.ipv4.ip_local_port_range=1024 65000
   - net.ipv4.conf.all.accept_redirects=0
   - net.ipv4.conf.all.secure_redirects=0
   - net.ipv4.conf.all.send_redirects=0
  labels:
   com.zabbix.description: "Zabbix server with PostgreSQL database support"
   com.zabbix.company: "Zabbix LLC"
   com.zabbix.component: "zabbix-server"
   com.zabbix.dbtype: "pgsql"
   com.zabbix.os: "alpine"

 zabbix-proxy-sqlite3:
  image: zabbix/zabbix-proxy-sqlite3:alpine-5.2-latest
  ports:
   - "10061:10051"
  volumes:
   - /etc/localtime:/etc/localtime:ro
   - /etc/timezone:/etc/timezone:ro 
   - ./zbx_env/usr/lib/zabbix/externalscripts:/usr/lib/zabbix/externalscripts:ro
   - ./zbx_env/var/lib/zabbix/modules:/var/lib/zabbix/modules:ro
   - ./zbx_env/var/lib/zabbix/enc:/var/lib/zabbix/enc:ro
   - ./zbx_env/var/lib/zabbix/ssh_keys:/var/lib/zabbix/ssh_keys:ro
   - ./zbx_env/var/lib/zabbix/mibs:/var/lib/zabbix/mibs:ro
   - snmptraps:/var/lib/zabbix/snmptraps:rw
  links:
   - zabbix-server:zabbix-server
   - zabbix-java-gateway:zabbix-java-gateway
  ulimits:
   nproc: 65535
   nofile:
    soft: 20000
    hard: 40000
  deploy:
   resources:
    limits:
      cpus: '0.70'
      memory: 512M
    reservations:
      cpus: '0.3'
      memory: 256M
  env_file:
   - .env_prx
   - .env_prx_sqlite3
  depends_on:
   - zabbix-java-gateway
   - zabbix-snmptraps
  networks:
   zbx_net_backend:
    aliases:
     - zabbix-proxy-sqlite3
     - zabbix-proxy-alpine-sqlite3
     - zabbix-proxy-sqlite3-alpine
   zbx_net_frontend:
  stop_grace_period: 30s
  labels:
   com.zabbix.description: "Zabbix proxy with SQLite3 database support"
   com.zabbix.company: "Zabbix LLC"
   com.zabbix.component: "zabbix-proxy"
   com.zabbix.dbtype: "sqlite3"
   com.zabbix.os: "alpine"

 zabbix-proxy-mysql:
  image: zabbix/zabbix-proxy-mysql:alpine-5.2-latest
  ports:
   - "10071:10051"
  volumes:
   - /etc/localtime:/etc/localtime:ro
   - /etc/timezone:/etc/timezone:ro 
   - ./zbx_env/usr/lib/zabbix/externalscripts:/usr/lib/zabbix/externalscripts:ro
   - ./zbx_env/var/lib/zabbix/modules:/var/lib/zabbix/modules:ro
   - ./zbx_env/var/lib/zabbix/enc:/var/lib/zabbix/enc:ro
   - ./zbx_env/var/lib/zabbix/ssh_keys:/var/lib/zabbix/ssh_keys:ro
   - ./zbx_env/var/lib/zabbix/mibs:/var/lib/zabbix/mibs:ro
   - snmptraps:/var/lib/zabbix/snmptraps:rw
  links:
   - zabbix-server:zabbix-server
   - zabbix-java-gateway:zabbix-java-gateway
  ulimits:
   nproc: 65535
   nofile:
    soft: 20000
    hard: 40000
  deploy:
   resources:
    limits:
      cpus: '0.70'
      memory: 512M
    reservations:
      cpus: '0.3'
      memory: 256M
  env_file:
   - .env_db_mysql_proxy
   - .env_prx
   - .env_prx_mysql
  secrets:
   - MYSQL_USER
   - MYSQL_PASSWORD
   - MYSQL_ROOT_PASSWORD
  depends_on:
   - mysql-server
   - zabbix-java-gateway
   - zabbix-snmptraps
  networks:
   zbx_net_backend:
    aliases:
     - zabbix-proxy-mysql
     - zabbix-proxy-alpine-mysql
     - zabbix-proxy-mysql-alpine
   zbx_net_frontend:
  stop_grace_period: 30s
  labels:
   com.zabbix.description: "Zabbix proxy with MySQL database support"
   com.zabbix.company: "Zabbix LLC"
   com.zabbix.component: "zabbix-proxy"
   com.zabbix.dbtype: "mysql"
   com.zabbix.os: "alpine"

 zabbix-web-apache-pgsql:
  image: zabbix/zabbix-web-apache-pgsql:alpine-5.2-latest
  ports:
   - "80:8080"
   - "443:8443"
  links:
   - postgres-server:postgres-server
   - zabbix-server:zabbix-server
  volumes:
   - /etc/localtime:/etc/localtime:ro
   - /etc/timezone:/etc/timezone:ro
   - ./zbx_env/etc/ssl/apache2:/etc/ssl/apache2:ro
   - ./zbx_env/usr/share/zabbix/modules/:/usr/share/zabbix/modules/:ro
#   - ./.ZBX_DB_CA_FILE:/run/secrets/root-ca.pem:ro
#   - ./.ZBX_DB_CERT_FILE:/run/secrets/client-cert.pem:ro
#   - ./.ZBX_DB_KEY_FILE:/run/secrets/client-key.pem:ro
  deploy:
   resources:
    limits:
      cpus: '0.70'
      memory: 512M
    reservations:
      cpus: '0.5'
      memory: 256M
  env_file:
   - .env_db_pgsql
   - .env_web
  secrets:
   - POSTGRES_USER
   - POSTGRES_PASSWORD
  depends_on:
   - postgres-server
   - zabbix-server
  healthcheck:
   test: ["CMD", "curl", "-f", "http://localhost:8080/"]
   interval: 10s
   timeout: 5s
   retries: 3
   start_period: 30s
  networks:
   zbx_net_backend:
    aliases:
     - zabbix-web-apache-pgsql
     - zabbix-web-apache-alpine-pgsql
     - zabbix-web-apache-pgsql-alpine
   zbx_net_frontend:
  stop_grace_period: 10s
  sysctls:
   - net.core.somaxconn=65535
  labels:
   com.zabbix.description: "Zabbix frontend on Apache web-server with PostgreSQL database support"
   com.zabbix.company: "Zabbix LLC"
   com.zabbix.component: "zabbix-frontend"
   com.zabbix.webserver: "apache2"
   com.zabbix.dbtype: "pgsql"
   com.zabbix.os: "alpine"

 zabbix-web-nginx-pgsql:
  image: zabbix/zabbix-web-nginx-pgsql:alpine-5.2-latest
  ports:
   - "8081:8080"
   - "8443:8443"
  links:
   - postgres-server:postgres-server
   - zabbix-server:zabbix-server
  volumes:
   - /etc/localtime:/etc/localtime:ro
   - /etc/timezone:/etc/timezone:ro
   - ./zbx_env/etc/ssl/nginx:/etc/ssl/nginx:ro
   - ./zbx_env/usr/share/zabbix/modules/:/usr/share/zabbix/modules/:ro
#   - ./.ZBX_DB_CA_FILE:/run/secrets/root-ca.pem:ro
#   - ./.ZBX_DB_CERT_FILE:/run/secrets/client-cert.pem:ro
#   - ./.ZBX_DB_KEY_FILE:/run/secrets/client-key.pem:ro
  deploy:
   resources:
    limits:
      cpus: '0.70'
      memory: 512M
    reservations:
      cpus: '0.5'
      memory: 256M
  env_file:
   - .env_db_pgsql
   - .env_web
  secrets:
   - POSTGRES_USER
   - POSTGRES_PASSWORD
  depends_on:
   - postgres-server
   - zabbix-server
  healthcheck:
   test: ["CMD", "curl", "-f", "http://localhost:8080/"]
   interval: 10s
   timeout: 5s
   retries: 3
   start_period: 30s
  networks:
   zbx_net_backend:
    aliases:
     - zabbix-web-nginx-pgsql
     - zabbix-web-nginx-alpine-pgsql
     - zabbix-web-nginx-pgsql-alpine
   zbx_net_frontend:
  stop_grace_period: 10s
  sysctls:
   - net.core.somaxconn=65535
  labels:
   com.zabbix.description: "Zabbix frontend on Nginx web-server with PostgreSQL database support"
   com.zabbix.company: "Zabbix LLC"
   com.zabbix.component: "zabbix-frontend"
   com.zabbix.webserver: "nginx"
   com.zabbix.dbtype: "pgsql"
   com.zabbix.os: "alpine"

 zabbix-agent:
  image: zabbix/zabbix-agent:alpine-5.2-latest
  ports:
   - "10050:10050"
  volumes:
   - /etc/localtime:/etc/localtime:ro
   - /etc/timezone:/etc/timezone:ro
   - ./zbx_env/etc/zabbix/zabbix_agentd.d:/etc/zabbix/zabbix_agentd.d:ro
   - ./zbx_env/var/lib/zabbix/modules:/var/lib/zabbix/modules:ro
   - ./zbx_env/var/lib/zabbix/enc:/var/lib/zabbix/enc:ro
   - ./zbx_env/var/lib/zabbix/ssh_keys:/var/lib/zabbix/ssh_keys:ro
  links:
   - zabbix-server:zabbix-server
  deploy:
   resources:
    limits:
      cpus: '0.2'
      memory: 128M
    reservations:
      cpus: '0.1'
      memory: 64M
   mode: global
  env_file:
   - .env_agent
  privileged: true
  pid: "host"
  networks:
   zbx_net_backend:
    aliases:
     - zabbix-agent
     - zabbix-agent-passive
     - zabbix-agent-alpine
  stop_grace_period: 5s
  labels:
   com.zabbix.description: "Zabbix agent"
   com.zabbix.company: "Zabbix LLC"
   com.zabbix.component: "zabbix-agentd"
   com.zabbix.os: "alpine"

 zabbix-java-gateway:
  image: zabbix/zabbix-java-gateway:alpine-5.2-latest
  ports:
   - "10052:10052"
  deploy:
   resources:
    limits:
      cpus: '0.5'
      memory: 512M
    reservations:
      cpus: '0.25'
      memory: 256M
  env_file:
   - .env_java
  networks:
   zbx_net_backend:
    aliases:
     - zabbix-java-gateway
     - zabbix-java-gateway-alpine
  stop_grace_period: 5s
  labels:
   com.zabbix.description: "Zabbix Java Gateway"
   com.zabbix.company: "Zabbix LLC"
   com.zabbix.component: "java-gateway"
   com.zabbix.os: "alpine"

 zabbix-snmptraps:
  image: zabbix/zabbix-snmptraps:alpine-5.2-latest
  ports:
   - "162:1162/udp"
  volumes:
   - snmptraps:/var/lib/zabbix/snmptraps:rw
  deploy:
   resources:
    limits:
      cpus: '0.5'
      memory: 256M
    reservations:
      cpus: '0.25'
      memory: 128M
  networks:
   zbx_net_frontend:
    aliases:
     - zabbix-snmptraps
   zbx_net_backend:
  stop_grace_period: 5s
  labels:
   com.zabbix.description: "Zabbix snmptraps"
   com.zabbix.company: "Zabbix LLC"
   com.zabbix.component: "snmptraps"
   com.zabbix.os: "alpine"

 mysql-server:
  image: mysql:8.0
  command:
   - mysqld
   - --character-set-server=utf8
   - --collation-server=utf8_bin
   - --default-authentication-plugin=mysql_native_password
  volumes:
   - ./zbx_env/var/lib/mysql:/var/lib/mysql:rw
  env_file:
   - .env_db_mysql
  secrets:
   - MYSQL_USER
   - MYSQL_PASSWORD
   - MYSQL_ROOT_PASSWORD
  stop_grace_period: 1m
  networks:
   zbx_net_backend:
    aliases:
     - mysql-server
     - zabbix-database
     - mysql-database

 postgres-server:
  image: postgres:alpine
#  command: -c ssl=on -c ssl_cert_file=/run/secrets/server-cert.pem -c ssl_key_file=/run/secrets/server-key.pem -c ssl_ca_file=/run/secrets/root-ca.pem
  volumes:
   - ./zbx_env/var/lib/postgresql/data:/var/lib/postgresql/data:rw
   - ./.ZBX_DB_CA_FILE:/run/secrets/root-ca.pem:ro
   - ./.ZBX_DB_CERT_FILE:/run/secrets/server-cert.pem:ro
   - ./.ZBX_DB_KEY_FILE:/run/secrets/server-key.pem:ro
  env_file:
   - .env_db_pgsql
  secrets:
   - POSTGRES_USER
   - POSTGRES_PASSWORD
  stop_grace_period: 1m
  networks:
   zbx_net_backend:
    aliases:
     - postgres-server
     - pgsql-server
     - pgsql-database

 db_data_mysql:
  image: busybox
  volumes:
   - ./zbx_env/var/lib/mysql:/var/lib/mysql:rw

 db_data_pgsql:
  image: busybox
  volumes:
   - ./zbx_env/var/lib/postgresql/data:/var/lib/postgresql/data:rw

# elasticsearch:
#  image: elasticsearch
#  environment:
#   - transport.host=0.0.0.0
#   - discovery.zen.minimum_master_nodes=1
#  networks:
#   zbx_net_backend:
#    aliases:
#     - elasticsearch

networks:
  zbx_net_frontend:
    driver: bridge
    driver_opts:
      com.docker.network.enable_ipv6: "false"
    ipam:
      driver: default
      config:
      - subnet: 172.16.238.0/24
  zbx_net_backend:
    driver: bridge
    driver_opts:
      com.docker.network.enable_ipv6: "false"
    internal: true
    ipam:
      driver: default
      config:
      - subnet: 172.16.239.0/24

volumes:
  snmptraps:

secrets:
  MYSQL_USER:
    file: ./.MYSQL_USER
  MYSQL_PASSWORD:
    file: ./.MYSQL_PASSWORD
  MYSQL_ROOT_PASSWORD:
    file: ./.MYSQL_ROOT_PASSWORD
  POSTGRES_USER:
    file: ./.POSTGRES_USER
  POSTGRES_PASSWORD:
    file: ./.POSTGRES_PASSWORD
```

Based on the docker-compose.yaml file, it should expose port 10050:10050 for zabbix-agent. I entered the Zabbix agent container and found that there is indeed something binding on 0.0.0.0:10050. But when I run `docker ps`, I don't see the ports exposed. I'm running it on Ubuntu 18.04 LTS using Docker.

I changed ZBX_SERVER_NAME to make it recognizable as my Zabbix installation instead of `Composed installation`.

There are the logs from the zabbix-agent container though I'm not sure if they are helpful.
```
 22264:20210318:122605.077 no active checks on server [zabbix-server:10051]: host [722372e65167] not found
 22264:20210318:122805.096 no active checks on server [zabbix-server:10051]: host [722372e65167] not found
 22264:20210318:123005.109 no active checks on server [zabbix-server:10051]: host [722372e65167] not found
 22264:20210318:123205.124 no active checks on server [zabbix-server:10051]: host [722372e65167] not found
```

**Comment by rockaut** [comment]
Ok ... sorry ZBX_SERVER_NAME is of course no problem.

First you should clean up your docker file - for example even as you're using pgsql as DB the template is also starting mysql and also multiple proxies (https://github.com/zabbix/zabbix-docker/wiki/FAQnP-...-Frequently-asked-questions-and-problems).

The agent is logging the host not found error because it has a dynamic hostname each time the container starts. You might want to use "hostname: zabbix-agent" for the service so it's fixed - there are some implications and situations where you don't want this that's why it isn't the default I guess.

Your problem still seam to be the same as in the other issue nonetheless I would say. Have you tried to create a new host in Zabbix with the Agent Template and "zabbix-agent" as dns? So this will do as passive agent. Then also remove the Agent Template from "Zabbix server" as this is tailored to the "non-container" world.

**Comment by aqiank** [comment]
Thank you. The Zabbix server finally can reach the agent after I added `hostname: zabbix-agent` to the docker-compose's zabbix-agent section. I had to stop + remove the agent container and restart it again with `docker-compose up -d` for it take effect. After that I followed your instruction to update the host to use DNS for the Zabbix agent in the web UI instead of the default `127.0.0.1` with IP and then it seems to work.
